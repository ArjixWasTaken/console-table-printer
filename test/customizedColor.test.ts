import { Table } from '../index';

describe('Example: Print a simple Table with Custom column colors', () => {
  it('Custom column colors are working', () => {
    // Create a table
    const p = new Table({
      columns: [
        { name: 'red_left_align_index', alignment: 'left', color: 'red' },
        { name: 'right_align_text', alignment: 'right' },
        { name: 'green_value', color: 'green' },
      ],
      colorMap: {
        red: '\x1b[32m', // actually its green
      },
    });

    // add rows with color
    p.addRow(
      {
        red_left_align_index: 2,
        right_align_text: 'This is looking green, but thats what I call red',
        green_value: 10.212,
      },
      { color: 'red' }
    );
    p.addRow(
      {
        red_left_align_index: 3,
        right_align_text: 'This row is blue as well',
        green_value: 10.212,
      },
      { color: 'blue' }
    );
    p.addRow(
      {
        red_left_align_index: 4,
        right_align_text: 'This row is green',
        green_value: 10.212,
      },
      { color: 'green' }
    );

    // print
    p.printTable();
    expect(p.render()).toMatchSnapshot();
  });
});
