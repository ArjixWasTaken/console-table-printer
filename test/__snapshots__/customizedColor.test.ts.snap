// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Example: Print a simple Table with Custom column colors Custom column colors are working 1`] = `
"┌──────────────────────┬──────────────────────────────────────────────────┬─────────────┐
[37m│[0m[37m [0m[01mred_left_align_index[0m[37m │[0m[37m [0m[01m                                right_align_text[0m[37m │[0m[37m [0m[01mgreen_value[0m[37m │[0m
├──────────────────────┼──────────────────────────────────────────────────┼─────────────┤
[37m│[0m[37m [0m[32m2                   [0m[37m │[0m[37m [0m[32mThis is looking green, but thats what I call red[0m[37m │[0m[37m [0m[32m     10.212[0m[37m │[0m
[37m│[0m[37m [0m[32m3                   [0m[37m │[0m[37m [0m[34m                        This row is blue as well[0m[37m │[0m[37m [0m[32m     10.212[0m[37m │[0m
[37m│[0m[37m [0m[32m4                   [0m[37m │[0m[37m [0m[32m                               This row is green[0m[37m │[0m[37m [0m[32m     10.212[0m[37m │[0m
└──────────────────────┴──────────────────────────────────────────────────┴─────────────┘"
`;
